src/components/utils/onfidoApi.ts

```diff
@@ -218,6 +222,79 @@ export const requestChallenges = (
   performHttpReq(options, onSuccess, (request) => formatError(request, onError))
 }
 
+/* v4 APIs */
+export const uploadBinaryMedia = (
+  { file, filename, sdkMetadata }: UploadDocumentPayload,
+  url: string | undefined,
+  token: string | undefined,
+  includeHmacAuth = false
+): Promise<UploadBinaryMediaReponse> =>
+  new Promise((resolve, reject) => {
+    try {
+      const tokenData = parseJwt(token)
+      const formData = new FormData()
+      formData.append('media', file, filename)
+      formData.append('sdk_metadata', JSON.stringify(sdkMetadata))
+
+      if (!includeHmacAuth) {
+        const requestParams: HttpRequestParams = {
+          endpoint: `${url}/v4/binary_media`,
+          payload: formData,
+          token: `Bearer ${token}`,
+        }
+
+        performHttpReq(requestParams, resolve, (request) =>
+          formatError(request, reject)
+        )
+
+        return
+      }
+
+      file
+        .arrayBuffer()
+        .then((data) => hmac256(tokenData.uuid as string, data))
+        .then((hmac) => {
+          const requestParams: HttpRequestParams = {
+            endpoint: `${url}/v4/binary_media`,
+            headers: { 'X-Video-Auth': hmac },
+            payload: formData,
+            token: `Bearer ${token}`,
+          }
+
+          performHttpReq(requestParams, resolve, (request) =>
+            formatError(request, reject)
+          )
+        })
+        .catch(reject)
+    } catch (error) {
+      reject(error)
+    }
+  })
+
+export const createV4Document = (
```

Comment:
How about calling this function `createMultiMediaDocument`

Reply:
That's a good name, but I don't think it's a good choice here. The design of `Document` was massively changed from `v3` to `v4`, which led to the changes in the document creation process. The term "multimedia document" doesn't fully reflect those changes, but rather introduce a new term that wasn't used anywhere else. IMO we can consider refactoring the current `uploadDocument` to `uploadV3Document` for better differentiation between `v3` & `v4` documents

Reply:
I see. IMO I would focus on what the function does rather then the API version it calls

Reply:
Yep, it should. But because the differences between 2 versions of `Document` were massive, it's kind of difficult to find a concise name for it :joy: